### PR TITLE
Increase test coverage: Array.prototype.toLocaleString

### PR DESCRIPTION
**Pull request by matedabis**
The coverage is measured by the following script, using `--jerry-test-suite --test262 --unittests --jerry-tests`:
https://github.com/matedabis/jerryscript/blob/gcov_coverage_tester/tests/gcov-tests/gcovtester.py

Branch coverage:
Before: 15/18
After: 18/18

JerryScript-DCO-1.0-Signed-off-by: Mate Dabis mdabis@inf.u-szeged.hu

### CODE REVIEW

File: tests/jerry/array-prototype-tolocalestring.js [hunk 1]
```diff
@@ -54,3 +54,33 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when null is passed to the function
+try {
+  Array.prototype.toLocaleString.call(null);
```
Comment (rerobika) [inline tests/jerry/array-prototype-tolocalestring.js]: Add an `assert (false)` after this call.

File: tests/jerry/array-prototype-tolocalestring.js [hunk 2]
```diff
@@ -54,3 +54,33 @@ try {
...
+}
+
+// Checking behavior when null is passed to the function
+try {
+  var obj = {};
+  Object.defineProperty(obj, 'length', { 'get' : function () { throw new ReferenceError ("foo"); } });
+  Array.prototype.toLocaleString.call(obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
```
Comment (rerobika) [inline tests/jerry/array-prototype-tolocalestring.js]: Please check the message as well! (And use unique messages)

File: tests/jerry/array-prototype-tolocalestring.js [hunk 3]
```diff
@@ -54,3 +54,33 @@ try {
...
+}
+
+// Checking behavior when length is an object, wichs valueOf throws error
```
Comment (rerobika) [inline tests/jerry/array-prototype-tolocalestring.js]: typo: `wichs`

**Comment by matedabis** [comment]
Corrected according to the reviews.
Replying to this comment: https://github.com/pando-project/jerryscript/pull/2716#discussion_r249092528
in the get function, and which is being catched are not the same `ReferenceError`s, as I check the message, I get assertion fail.

File: tests/jerry/array-prototype-tolocalestring.js [hunk 4]
```diff
@@ -54,3 +54,34 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when null is passed to the function
```
Comment (rerobika) [inline tests/jerry/array-prototype-tolocalestring.js]: This description is a bit misleading. I'd suggest the same as in #2717.

**Comment by matedabis** [comment]
I updated the patch according to the review.

File: tests/jerry/array-prototype-tolocalestring.js (hunk 2 above)
Comment (matedabis) [reply tests/jerry/array-prototype-tolocalestring.js]: The thrown error and the one in the function are not the same `ReferenceError`s, so as I check the message, I get assertion fail.

Comment (akosthekiss) [reply tests/jerry/array-prototype-tolocalestring.js]: Something is not right then. Q1: Why set the error message then, if it cannot be checked? Q2: Does the error object get replaced somewhere by the builtin implementation? Q3: If yes, why? Q4: If yes, is it necessary to throw a reference error within the anonymous function or any kind of error can be thrown?

**Comment by matedabis** [comment]
Updated the patch:
- Added comments for each case containing which part of the ES 5.1 it tests.
- Ran the test code in Gecko, V8, SpiderMonkey engines and got the same result.

File: tests/jerry/array-prototype-tolocalestring.js (hunk 2 above)
Comment (matedabis) [reply tests/jerry/array-prototype-tolocalestring.js]: Sorry I probably made a typo, thats why it did not work. It works with the message check.

Comment (matedabis) [reply tests/jerry/array-prototype-tolocalestring.js]: Sorry I made a typo, that is why it did not work.

**Comment by matedabis** [comment]
Corrected the issue above.

**Comment by matedabis** [comment]
Closing PR because of moving to another branch.

**Pull request closed without merging**